ui-src/lib/views/DatePickerView.js

```diff
@@ -0,0 +1,257 @@
+   */
+  _initWithOptions(options = null) {
+    this._alive = true;
+    options =
+      options !== null
+        ? options
+        : {
+            disabled: true,
+          };
+    this._pickerElement.datepicker(options);
+  }
+
+  /**
+   * Execute a specified datepicker function, passing the arguments.
+   *
+   * This function exists to catch the cases where functions are called when
+   * the picker is not attached (alive).
+   *
+   * The first argument to this function should be the name of the desired
+   * datepicker function, followed by the correct arguments for that function.
+   *
+   * @return {Any} Function result
+   */
+  _apply() {
+    if (!this._alive) {
+      console.log('WARN: datepicker not constructed yet [' + this.domId + ']');
+    }
+    return this._pickerElement.datepicker(...arguments);
+  }
+
+  /**
+   * Activate the datepicker if not already attached.
+   *
+   * Basically calls _initWithOptions({disabled: disabled}), but only if not
+   * already attached. Otherwise just sets the disabled status.
+   *
+   * @param  {Boolean} disabled True if datepicker needs to be disabled
+   */
+  activate(disabled = true) {
+    if (this._alive) {
+      this.disabled = disabled;
+    } else {
+      this._initWithOptions({
+        disabled: disabled,
+      });
+    }
+  }
+
+  /**
+   * Get the element the datepicker is attached to.
+   *
+   * @return {jQuery} jQuery element
+   */
+  get element() {
+    return this._pickerElement;
+  }
+
+  /**
+   * Set option or options to the datepicker, like the 'option' call with
+   * various arguments.
+   *
+   * Special case is when the datepicker is not (yet) attached. In that case
+   * we need to initialze the datepicker with the options instead.
+   *
+   * @return {Any} Result of the 'option' call.
+   */
+  option() {
+    /*
+     * Special case the scenario of calling option setting with just a map of
+     * settings, when the picker is not alive. That really should translate
+     * to a constructor call to the datepicker.
+     */
+    if (
+      !this._alive &&
+      arguments.length == 1 &&
+      (arguments[0] === null || typeof arguments[0] === 'object')
+    ) {
+      console.log('DP special case  [' + this.domId + ']: ', arguments[0]);
+      return this._initWithOptions(arguments[0]);
+    }
+    return this._apply('option', ...arguments);
+  }
+
+  /**
+   * Return current set of options.
+   *
+   * This is special cased here vs. documentation. We need to ask for 'all'.
+   *
+   * @return {Object} Datepicker options
+   */
+  options() {
+    return this.option('all');
+  }
+
+  /**
+   * Determine whether datepicker is disabled.
+   *
+   * @return {Boolean} True if disabled.
+   */
+  get isDisabled() {
+    return this._apply('isDisabled');
+  }
+
+  /**
+   * Set dsiabled status of picker.
+   *
+   * @param  {Boolean} disabled True to disable
+   */
+  set disabled(dsiabled) {
```

Comment:
dsiabled?

I'm surprised I don't see an error for this. Is this function actually used?

Reply:
If you did not see an error it is because your execution path never reached it. Oh the joy of interpreted dynamic languages (we could switch to TypeScript to alleviate some of this).

You are right though in the sense that this function is not currently used. When I write classes like these, however, I try to make them functionally complete to avoid surprises later. Since this class encapsulates the jQuery date picker functionality and it has a disable function...

Fixed the spelling though.

Reply:
Hmm, I'm generally more of the YAGNI school of thought. And particularly in JavaScript where compilation doesn't catch this sort of bug, do we avoid surprises by checking in a bunch of never-tested, dead code?

Reply:
Re: YAGNI. I personally don't see a few "missing" proxy methods as lacking correct architecture. It's just boilerplate that can be added when needed. But I don't want to hold up the UI over this disagreement; if you're happier to have them, I'll let it go.

Re: testing: I understand the difficulty of setting up automated testing of UI code, and I'm not asking you to do so. (Though I'd be thrilled to have it!) By "never-tested, dead" I meant ever run in any way, even tested manually once through the Javascript console. (That's obvious in this case given that it didn't work; I assume some of these other methods are similarly untested.)